apps/admin/src/app/(api)/api/v1/anomaly/route.ts

```diff
@@ -0,0 +1,56 @@
+import { NextResponse, type NextRequest } from "next/server";
+import { isNoticeableEnergyConsumption } from "@/actions/energy-monitoring";
+import { env } from "@/env.mjs";
+import { getSensorsByUser } from "@/query/sensor";
+import { getAllUsers } from "@/query/user";
+
+import { getEnergyForSensorInRange } from "@energyleaf/db/query";
+import type { ConsumptionData } from "@energyleaf/lib";
+import { AggregationType } from "@energyleaf/lib";
+import { sendAnomalyEmail } from "@energyleaf/mail";
+
+export const POST = async (req: NextRequest) => {
+    const reportApiKey = env.REPORTS_API_KEY;
+    if (!req.headers.has("Authorization") || req.headers.get("Authorization") !== reportApiKey) {
+        return NextResponse.json({ status: 403, statusMessage: "Forbidden" });
+    }
+
+    const start = new Date(new Date().setHours(0, 0, 0, 0));
+    const end = new Date(new Date().setHours(23, 59, 59, 999));
+
+    try {
+        const users = await getAllUsers();
+        for (const user of users) {
+            const sensors = await getSensorsByUser(user.id);
+            let data: ConsumptionData[] = [];
+
+            for (const sensor of sensors) {
+                const energyData = await getEnergyForSensorInRange(start, end, sensor.id, AggregationType.RAW);
+                data = data.concat(
+                    energyData.map((entry) => ({
+                        sensorId: entry.sensorId || 0,
+                        energy: entry.value,
+                        timestamp: entry.timestamp.toString(),
+                    })),
+                );
+            }
+
+            if (!isNoticeableEnergyConsumption(data)) {
+                continue;
+            }
+
+            await sendAnomalyEmail({
+                to: user.email,
+                name: user.username,
+                from: env.RESEND_API_MAIL,
+                apiKey: env.RESEND_API_KEY,
+                unsubscribeLink: "",
+                link: "",
```

Comment:
Da weiß ich nicht, auf welche Seite genau der Link führen soll.

Reply:
Wir haben in irgendeinem Meeting mal besprochen, dass man über die E-Mail einen Link anklicken kann, um den ungewöhnlichen Verbrauch direkt auf dem Dashboard angezeigt zu bekommen. Ist das vllt damit gemeint?

Reply:
Also führt der Link einfach zu `energyleaf.de` oder hat das Dashboard noch eine extra Route?

Reply:
Ja, erstmal können wir den Link zu `energyleaf.de` machen. Die extra Route im Dashboard gibt es so noch nicht und muss erst noch implementiert werden